src/Rules/LibIMEJyutping.hs

```diff
@@ -0,0 +1,66 @@
+module Rules.LibIMEJyutping (libIMEJyutpingRule) where
+
+import Base
+
+fcitxDataUrl :: String
+fcitxDataUrl = "https://download.fcitx-im.org/data/"
+
+libIMEJyutpingRule :: Rules ()
+libIMEJyutpingRule = do
+  jyutpingToolsRule
+  jyutpingDictRule
+  jyutpingLmRule
+  "libime-jyutping" ~> do
+    copyFile' (outputDir </> "jyutping.dict") $ outputDir </> "libime-jyutping" </> "jyutping.dict"
+    copyFile' (outputDir </> "zh_HK.lm") $ outputDir </> "libime-jyutping" </> "zh_HK.lm"
+    copyFile' (outputDir </> "zh_HK.lm.predict") $ outputDir </> "libime-jyutping" </> "zh_HK.lm.predict"
+
+jyutpingToolsRule :: Rules ()
+jyutpingToolsRule = do
+  "libime-jyutping-tools" ~> do
+    need ["libime-tools"]
+    let libIMEJyutpingSrc = "libime-jyutping"
+    cmd_ "sed" "-i" "46{\\|find_package(Fcitx5Module|d}" (libIMEJyutpingSrc </> "CMakeLists.txt")
+    cmd_ "cmake" "-B" (libIMEJyutpingSrc </> "build") "-G" "Ninja"
+      [
+        "-DCMAKE_BUILD_TYPE=Release",
+        "-DCMAKE_INSTALL_PREFIX=" <> outputDir,
+        "-DLibIMECore_DIR=" <> outputDir </> "lib" </> "cmake" </> "LibIMECore",
+        "-DENABLE_TEST=OFF",
+        "-DENABLE_ENGINE=OFF"
+      ]
+      libIMEJyutpingSrc
+    cmd_ "cmake" "--build" (libIMEJyutpingSrc </> "build") "--target" "libime_jyutpingdict"
+    -- ignore install errors
+    Exit c <- cmd "cmake" "--install" (libIMEJyutpingSrc </> "build") "--component" "lib"
+    Exit c <- cmd "cmake" "--install" (libIMEJyutpingSrc </> "build") "--component" "tools"
+    cmd_ "echo" ("installed to " <> outputDir)
```

Comment:
Do we still need this?

Reply:
Yes, some tools are not built and would throw error when install. I'm tired of fiddling with cmake and let's just ignore those errors.

Reply:
I'm saying line 37, the echo command

Reply:
I don't know. If the last statement is not a `cmd_`, there would be a compile error:

```
The last statement in a 'do' block must be an expression
  Exit c <- cmd
```